api/src/main/java/org/xnio/XnioWorker.java

```diff
@@ -874,6 +877,12 @@ protected final int getWorkerQueueSize() {
         return taskQueue.size();
     }
 
+    public abstract XnioWorkerMXBean getMetrics();
+
+    protected abstract Closeable registerServerMetrics(XnioServerMXBean metrics);
```

Comment:
Instead of Closeable, let's introduce a new interface into the top XNIO API package that extends AutoCloseable but throws no exception.